### PR TITLE
migrate test_vmctl to pytest

### PR DESCRIPTION
**Pull request by frogunder**
### What does this PR do?

migrate test_vmctl to pytest

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/unit/modules/test_vmctl.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    """
+    Tests starting a VM that is already defined.
+    """
+    ret = {}
+    ret["stderr"] = "vmctl: started vm 4 successfully, tty /dev/ttyp4"
+    ret["retcode"] = 0
+    mock_cmd = MagicMock(return_value=ret)
+    expected = {"changes": True, "console": "/dev/ttyp4"}
+    with patch.dict(vmctl.__salt__, {"cmd.run_all": mock_cmd}):
+        assert expected == vmctl.start("4")
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_vmctl.py:124]:
```suggestion
        assert vmctl.start("4") == expected 
```

File: tests/pytests/unit/modules/test_vmctl.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                    "-i 2",
+                    "-b",
+                    "/bsd.rd",
+                    "-d",
+                    "/disk.img",
+                ],
+                output_loglevel="trace",
+                python_shell=False,
+            )
+            assert expected == res
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_vmctl.py:154]:
```suggestion
            assert res == expected
```

File: tests/pytests/unit/modules/test_vmctl.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            "maxmem": "512M",
+            "owner": "george",
+            "pid": "-",
+            "state": "stopped",
+            "tty": "-",
+            "vcpus": "1",
+        },
+    }
+    with patch.dict(vmctl.__salt__, {"cmd.run_all": mock_cmd}):
+        assert expected == vmctl.status()
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_vmctl.py:203]:
```suggestion
        assert vmctl.status() == expected 
```

File: tests/pytests/unit/modules/test_vmctl.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+            "maxmem": "2.9G",
+            "owner": "ringo",
+            "pid": "123",
+            "state": "running",
+            "tty": "ttyp5",
+            "vcpus": "1",
+        },
+    }
+    with patch.dict(vmctl.__salt__, {"cmd.run_all": mock_cmd}):
+        assert expected == vmctl.status("web4")
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_vmctl.py:231]: What we expect should be on the right side of `==`.

Let's clean all the asserts to match this.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_vmctl.py (hunk 4 above)
Comment (frogunder) [reply tests/pytests/unit/modules/test_vmctl.py]: Thanks Pedro, should be fixed now.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**